plugin/store.go

```diff
@@ -136,13 +136,13 @@ func (ps *Store) Get(name, capability string, mode int) (plugingetter.CompatPlug
 	// Lookup using legacy model.
 	if allowV1PluginsFallback {
```

Comment:
(Not sure, but thinking out loud): would it be an option to return early here as well?

```go
if !allowV1PluginsFallback {
    return nil, errNotFound(name)
}
```

Reply:
sounds good. I am not sure `errNotFound(name)` is good enough but lets go with that.

Reply:
Well, if we reached this point, and there was no error in the `if ps != nil {` branch above, then I guess it means the plugin was not found. (unless `ps` was nil, but I'm not really sure what case that would be; if that's invalid, we need an error for that)